### DISCUSSION TITLE
Rebuild destroyed buildings

### DISCUSSION THREAD

**Discussion by mathieumahe** [Keriew/augustus#342] [Ideas]
Hi,

It can happen that a part of a city is destroyed (by fire or enemies or ...) and you want to rebuild the destroyed buildings exactly the same.

You can right click on the ruins of a building to remember what it was, but you have lost the information about how it was configured. And you need to find the right building in the construction menus.

That's why I think that a button "rebuild this exact building" in the right-click window would improved the gamer life. What do you think?

**Comment by crudelios** [comment]
I think the idea is good, I'm not sure the game engine allows for this. Apart from the name, rubble tiles lose all remaining information about the destroyed building, mainly what its position in the city grid was. If you have many rubble tiles with similar buildings (i.e. housing) there's no way right now for the game to figure out where to restore the building.

**Reply by mathieumahe** [reply]
If the game remember the position and the name of the destroyed building, I guess it's possible to, at least, rebuild the same building, no ?
(But yeah, I understand that saving the way the building was configured before it was destroy is way more work!)

**Reply by crudelios** [reply]
What I meant is that the game doesn't store the building size anywhere when it is destroyed, so there's no way for a rubble tile to know what specific part of a multi-tile building it belonged to.

**Comment by Mephisto-Fibs** [comment]
If the rubble tile at least 'remembers' what it was when klicking on it's info window, there could be a button to klick in that window that selects the building in order for the player to place it again ON the rubble tiles... Then it must be visited by a worker from an architects guild to reappeare as a building

would that work?

**Comment by crudelios** [comment]
Now that I have a bit more understanding of the building code I think there can be a way to implement this going forward, but it'll go to my already large pile of to-dos 😉
